try:                                      # отлавливает ошибки пример
    1/int(input())
    print("поделилось")
except ZeroDivisionError:
    print("кто-то поделил на 0")
except ValueError:
    print("не получилось превратить в число - лучше указать конкретную ошибку из списка")
else:
    print("если не было ошибок")
finally:
    print("может пригодиться - например закрыть в любом случае")